### PR TITLE
recommend highlight api

### PR DESCRIPTION
**Pull request by sywhb**
- Add recommendHighlights API
- Recommend highlights
- Allow recommended With Highlights

### CODE REVIEW

File: packages/api/src/schema.ts [hunk 1]
```diff
@@ -2318,6 +2344,9 @@ const schema = gql`
     createGroup(input: CreateGroupInput!): CreateGroupResult!
     recommend(input: RecommendInput!): RecommendResult!
```
Comment (jacksonh) [inline packages/api/src/schema.ts:2345]: I wonder if its better to just extend the `recommend` API? like `includeHighlights: bool`. At the time of recommending, does the backend have access to all the highlights?

File: packages/api/src/generated/schema.graphql [hunk 2]
```diff
@@ -1248,16 +1249,40 @@ enum RecommendErrorCode {
...
+
+type RecommendHighlightsSuccess {
+  success: Boolean!
+}
+
 input RecommendInput {
   groupIds: [ID!]!
   note: String
   pageId: ID!
+  recommendedWithHighlights: Boolean
```
Comment (sywhb) [inline packages/api/src/generated/schema.graphql:1279]: @jacksonh There is `recommendedWithHighlights` in the `RecommendInput`

File: packages/api/src/schema.ts (hunk 1 above)
Comment (sywhb) [reply packages/api/src/schema.ts:2345]: We have both methods to recommend highlights.
Just want to be more flexible. You can just use the original `recommend` api if you want

File: packages/api/src/generated/schema.graphql (hunk 2 above)
Comment (jacksonh) [reply packages/api/src/generated/schema.graphql:1279]: Ah i see, so from the client perspective I would probably always use that option. Will probably add a toggle on the share box that people can use to enable it.

Comment (sywhb) [reply packages/api/src/generated/schema.graphql:1279]: So you want me to remove the recommendHighlight API?

Comment (jacksonh) [reply packages/api/src/generated/schema.graphql:1279]: @sywhb I don't think that we need it now, I can see how maybe in the future we might want to share a single highlight or something though.

Its probably ok to leave it in.

Comment (sywhb) [reply packages/api/src/generated/schema.graphql:1279]: Cool. We can merge it now

**Review by jacksonh — APPROVED** [review]

**Comment by jacksonh** [comment]
Going to merge this one so I can test it on demo

**Pull request merged**